### PR TITLE
Prettier

### PR DESCRIPTION
**Pull request by mattgrill**
- prettier
- run prettier

### CODE REVIEW

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "printWidth": 80,
+  "semi": true,
+  "singleQuote": true,
+  "trailingComma": "all"
```
Comment (dawehner) [inline .prettierrc.json]: I love trailing comma all!

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "extends": [
+    "eslint-config-airbnb",
+    "plugin:prettier/recommended"
```
Comment (dawehner) [inline .eslintrc.json]: I tried to compare what eslint-plugin-prettier does: https://github.com/prettier/eslint-plugin-prettier#recommended-configuration vs. what https://github.com/facebook/create-react-app/blob/next/packages/eslint-config-react-app/index.js does. Given we already try to go down the airbnb ruleset in core, I think using the same here makes 100% sense.

File: package.json [hunk 3]
```diff
@@ -9,9 +9,19 @@
...
+    "eslint-config-prettier": "^2.9.0",
+    "eslint-plugin-import": "^2.7.0",
+    "eslint-plugin-jsx-a11y": "^6.0.2",
+    "eslint-plugin-prettier": "^2.6.0",
+    "eslint-plugin-react": "^7.4.0",
+    "prettier": "^1.11.1",
     "react-scripts": "1.1.1"
   },
   "scripts": {
+    "prettier": "./node_modules/.bin/prettier --write 'src/**/*.jsx' 'src/**/*.js'",
```
Comment (dawehner) [inline package.json]: Can't you just use "prettier"? It'll pick up things in node_modules/.bin/prettier automatically

Comment (mattgrill) [reply package.json]: Yeah, you can. I just always try and be specific.

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  "parser": "babel-eslint",
+  "root": true,
+  "env": {
+    "browser": true,
+    "es6": true
+  },
+  "rules": {
+    "no-param-reassign": [0],
+    "no-underscore-dangle": [0],
+    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }]
```
Comment (justafish) [inline .eslintrc.json]: Let's ditch jsx completely

**Comment by mattgrill** [comment]
@justafish I just looked into this, if you add an .eslintrc.json it picks it up.

File: .eslintrc.json (hunk 4 above)
Comment (mattgrill) [reply .eslintrc.json]: We can, however there are some tests that are `.js` but they contain JSX. I don't care either way which is why I added this.

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
+  "extends": [
+    "eslint-config-airbnb",
+    "plugin:prettier/recommended"
+  ],
+  "parser": "babel-eslint",
+  "root": true,
+  "env": {
+    "browser": true,
+    "es6": true
```
Comment (justafish) [inline .eslintrc.json]: + jest, and serviceworker

**Comment by justafish** [comment]
Just the linting errors themselves that need fixing now :+1:

**Comment by justafish** [comment]
So one thing I don't understand about adding `eslint-config-prettier` is it says that it "Turns off all rules that are unnecessary or might conflict with Prettier." - so what is enforcing the rules that it's turned off? Running prettier is something that each person needs to have done manually (or setup to do automatically on save, but still). If that is the case, then I would have thought our prettier config would be setup to help you follow the standards we've adopted (airbnb in this case), and so I'm not sure what this plugin actually does?

**Comment by justafish** [comment]
Ok I read https://prettier.io/docs/en/eslint.html and I get it now :+1:

**Pull request merged**